IEEE-PSH/hackpsh-web
# Schema Additions & Server-side DB Fetching via tRPC

## Description

* Introduced table definition for announcements & teams & roles
* Created pattern for pre-loading db with seed values (`migrate.ts`)
* Handled redirect on sign-in and page traversal to force users to 
complete onboarding before platform experience
* Utilizing serverTRPC to communicate with db routes **(issue)**
  - Utilizing postgres library introduces issues with webpack using cloudflare:uri links due to an internal optional dependency within postgres causing module build / bundle issues.

## Bugfixes

* Prevent home route from being cached, allowing traversal back to dashboard without valid session credentials due to router cache (Cache Busting) via `router.refresh()`

## Test Guidelines

* Review supabase - staging environment to see DB Changes have been applied successfully with pre-seed values
* Repro auth flow cache issue (login -> home -> dashboard -> logout -> eval session button = "Login" & back button)
